R/cal-estimate-utils.R

```diff
@@ -194,27 +194,23 @@ tidyselect_cols <- function(.data, x) {
 # any tidyeval variable calls made prior to calling the calibration
 split_dplyr_groups <- function(.data) {
   if (dplyr::is_grouped_df(.data)) {
-    .data %>%
-      dplyr::summarise(.groups = "drop") %>%
-      purrr::transpose() %>%
-      purrr::map(~ {
-        purrr::imap(.x, ~ expr(!!parse_expr(.y) == !!.x)) %>%
-          purrr::reduce(function(x, y) expr(!!x & !!y))
-      }) %>%
-      purrr::map(~ {
-        df <- .data %>%
-          dplyr::filter(, !!.x) %>%
-          dplyr::ungroup()
-
-        list(
-          data = df,
-          filter = .x,
-          rows = nrow(df)
-        )
-      })
+    grp_keys <- .data %>% dplyr::group_keys()
+    grp_var <- .data %>% dplyr::group_vars()
+    grp_sym <- rlang::sym(grp_var)
+    grp_data <- .data %>% tidyr::nest()
+    grp_filters <-
+      purrr::map(grp_keys[[1]], ~ expr(!!parse_expr(grp_var) == !!.x))
```

Comment:
See the comment below. I think we constrained it to a single grouping column because the plotting functions wouldn't work well with multiple groupings. 

I'll commit this because it is a good idea but put some error-checking code in so that we fail with a good error if there are multiple grouping columns.